File: python/300.py
class Solution:
    def lengthOfLIS(self, nums: List[int]) -> int:
        if nums is None or len(nums) == 0:
            return 0
        DP = [ 1 for _ in range(len(nums))]
        result = 1
        for i in range(1, len(nums)):
            for j in range(0, i):
                if nums[i] > nums[j]:
                    DP[i] = max(DP[i], DP[j] + 1)
            result = max(result, DP[i])
        return result

File: java/300.java
class Solution {
    public int lengthOfLIS(int[] nums) {
        if (nums == null || nums.length == 0)
            return 0;
        int[] DP = new int[nums.length];
        int result = 1;
        for (int i = 0; i < DP.length; i++) {
            DP[i] = 1;
        }
        for (int i = 1; i < nums.length; i++) {
            for (int j = 0; j < i; j++) {
                if (nums[j] < nums[i]) {
                    DP[i] = Math.max(DP[i], DP[j] + 1);
                }
            }
            if (DP[i] > result) result = DP[i];
        }
        return result;
    }
}
